pkg/knowledge_base2/properties/resource_property.go

```diff
@@ -0,0 +1,164 @@
+package properties
+
+import (
+	"fmt"
+
+	"github.com/klothoplatform/klotho/pkg/collectionutil"
+	construct "github.com/klothoplatform/klotho/pkg/construct2"
+	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base2"
+)
+
+type (
+	ResourceProperty struct {
+		DefaultValue *construct.ResourceId  `json:"default_value" yaml:"default_value"`
+		AllowedTypes construct.ResourceList `yaml:"allowed_types"`
+		Namespace    bool                   `yaml:"namespace"`
+		SharedPropertyFields
+		*knowledgebase.PropertyDetails
```

Comment:
Embedded fields that are pointers are a little weird and could more easily lead to nil-pointer access. Is there a reason this needs to be a pointer?

Reply:
flipped we chatted about this